Overhaul AI Gateway docs for new architecture (#2196)
ngrok/ngrok-docs: ai-gateway/custom-providers/vllm.mdx

description: Route AI requests to vLLM inference servers through the ngrok AI Gateway.
---

[vLLM](https://github.com/vllm-project/vllm) is a high-performance inference engine with an OpenAI-compatible API. Connect your vLLM server to the AI Gateway as a [custom provider](/ai-gateway/concepts/custom-providers).

## Prerequisites

- [ngrok account](https://app.ngrok.ai) with AI Gateway access
- [vLLM](https://docs.vllm.ai/en/latest/getting_started/installation.html) installed
- [ngrok agent](https://download.ngrok.com) installed
- GPU with sufficient VRAM for your chosen model
- An [access key](/ai-gateway/concepts/api-keys) from [app.ngrok.ai](https://app.ngrok.ai)

## Overview

> vLLM exposes an OpenAI-compatible server. Expose it with an ngrok internal endpoint (or use a public HTTPS URL), register it as a custom provider, and attach credentials through an access key configuration.

Reviewer: prefer to avoid "exposing" in back to back sentences 😁

```suggestion
vLLM provides an OpenAI-compatible server that you can expose with an ngrok internal endpoint (or use a public HTTPS URL), register as a custom provider, and attach credentials through an access key configuration.
```
